cmd/kubeadm/app/cmd/reset.go

```diff
@@ -55,6 +55,12 @@ func NewCmdReset(in io.Reader, out io.Writer) *cobra.Command {
 			ignorePreflightErrorsSet, err := validation.ValidateIgnorePreflightErrors(ignorePreflightErrors)
 			kubeadmutil.CheckErr(err)
 
+			if criSocketPath == "" {
+				criSocketPath, err = utilruntime.DetectCRISocket()
```

Comment:
I think that the detection process should initially try to read the kubeadm config (that now includes also the node annotation with the cri socket used for node initialisation) and only as a fallback use cri detection.

This has two advantages:
1) it will work also in case for multiple cri (because the cri to use is already known)
2) it will enable further cleanup actions e.g. cleanup of cluster status in case of control plane nodes (out of scope of this PR)